lib/renderer.js

```diff
@@ -92,9 +92,19 @@ Renderer.prototype.link = function(href, title, text) {
     out += ` title="${title}"`;
   }
 
+  const target = ' target="_blank"'
+  const noopener = ' rel="noopener"';
+  const nofollowTag = ' rel="noopener external nofollow noreferrer"';
+
   if (external_link) {
-    if (external_link.enable && isExternal(href, options.config, external_link.exclude)) {
-      out += ' target="_blank" rel="noopener"';
+    if (isExternal(href, options.config, external_link.exclude)) {
+      if (external_link.enable && external_link.nofollow) {
+        out += target + nofollowTag;
+      } else if (external_link.enable) {
+        out += target + noopener;
+      } else if (external_link.nofollow) {
+        out += nofollowTag;
+      }
```

Comment:
I have an idea to simplify it:

```
  const target = ' target="_blank"'
  const rel = [];
```

```js
if (isExternal(href, options.config, external_link.exclude)) {
  if (external_link.enable) {
    out += target;
    rel.push('noopener');
  }
  if (external_link.nofollow) {
    rel.push(...['noopener', 'external', 'nofollow', 'noreferrer']);
  }

  out += `rel="${[...new Set(rel)].join(' ')}"`;
}
```

Reply:
Aside from `isExternal` (which is still pending https://github.com/hexojs/hexo-util/pull/133), I'm not convinced using array is simpler in this case.

`+` operator is [very fast](https://stackoverflow.com/a/55409094) and more readable.

Reply:
The approach I gave is to slightly reduce the cyclomatic complexity. But apparently, the current one i s definitely more readable. 
Details: https://eslint.org/docs/rules/complexity